utilities/xmlrpcpp/src/XmlRpcDispatch.cpp

```diff
@@ -132,11 +134,12 @@ XmlRpcDispatch::work(double timeout)
       // Only handle requested events to avoid being prematurely removed from dispatch
       bool readable = (pfd.events & POLLIN_REQ) == POLLIN_REQ;
       bool writable = (pfd.events & POLLOUT_REQ) == POLLOUT_REQ;
+      bool oob = (pfd.events & POLLEX_REQ) == POLLEX_REQ;
       if (readable && (pfd.revents & POLLIN_CHK))
         newMask &= src->handleEvent(ReadableEvent);
       if (writable && (pfd.revents & POLLOUT_CHK))
         newMask &= src->handleEvent(WritableEvent);
-      if (pfd.revents & POLLEX_CHK)
+      if (oob && (pfd.revents & POLLEX_CHK))
         newMask &= src->handleEvent(Exception);
```

Comment:
Can you describe more about what is going on here? Is this a behaviour change/fix which arose as a result of the added tests?

Reply:
I initially wrote this test against the previous version of this, which was using `select()`, since our internal fork of ROS is mostly based on Indigo. When I updated this test to the most recent version of ros_comm to get the `poll()` implementation, it no longer behaved the same way when there was an exception event on the socket. This fixes it so that the `poll()` implementation behaves more like the original `select()` implementation.

Also of note is that for TCP sockets, the exception event actually indicates the reception of out-of-band data over the TCP connection, and doesn't indicate an error event on the socket. I've updated the comments and the implementation to reflect this and restore the original behavior.

(I light of this I think the original behavior might be a little misguided, but my goal here is to preserve all of the existing behavior that isn't clearly a bug.)

Reply:
Fun, okay. Yeah, we're having some similar issues with select/poll/epoll subtleties over in #1217.